packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -230,9 +240,21 @@ export default function MainFeedLayout({
     if (!query.query) {
       return null;
     }
-    const variables = isUpvoted
-      ? { ...query.variables, period: periods[selectedPeriod].value }
-      : query.variables;
+
+    const getVariables = () => {
+      if (isUpvoted) {
+        return { ...query.variables, period: periods[selectedPeriod].value };
+      }
+
+      if (sortingEnabled && isSortableFeed) {
```

Comment:
I'd suggest this change. Otherwise, once `sortingEnabled` will be turned on it will refresh the feed.
In any way, the default is by popularity so it should make any different

```suggestion
      if (isSortableFeed) {
```

Reply:
Not sure I follow. We don't save the selected sorting method locally, the user has to explicitly choose it every time so when `selectedAlgo` is changed we need to trigger our `useMemo` again. So just add it to the deps list of the `useMemo`